service_capacity_modeling/models/org/netflix/ddb.py

```diff
@@ -0,0 +1,584 @@
+) -> float:
+    mean_item_size = extra_model_arguments.get("estimated_mean_item_size_bytes", 0)
+    if mean_item_size == 0:
+        mean_item_size = desires.query_pattern.estimated_mean_write_size_bytes.mid
+    return mean_item_size
+
+
+def _get_num_regions(extra_model_arguments: Dict[str, Any]) -> int:
+    return extra_model_arguments.get("number_of_regions", 1)
+
+
+def _get_dynamo_standard(
+    context: RegionContext, extra_model_arguments: Dict[str, Any]
+) -> Service:
+    number_of_regions = _get_num_regions(extra_model_arguments)
+    dynamo_service = (
+        context.services.get("dynamo.standard.global", None)
+        if number_of_regions > 1
+        else context.services.get("dynamo.standard", None)
+    )
+    if not dynamo_service:
+        raise ValueError("DynamoDB Service is not available in context")
+    return dynamo_service
+
+
+def _get_dynamo_backup(context: RegionContext) -> Service:
+    dynamo_backup = context.services.get("dynamo.backup.continuous", None)
+    if not dynamo_backup:
+        raise ValueError("DynamoDB Backup is not available in context")
+    return dynamo_backup
+
+
+def _get_dynamo_transfer(
+    context: RegionContext,
+) -> List[Tuple[float, float]]:
+    transfer_costs = context.services.get("dynamo.transfer", None)
+    if not transfer_costs:
+        raise ValueError(
+            "DynamoDB Transfer cost is not available in context"
+        )
+    return transfer_costs.annual_cost_per_gib
+
+
+def _plan_writes(
+    context: RegionContext,
+    desires: CapacityDesires,
+    extra_model_arguments: Dict[str, Any],
+) -> _WritePlan:
+    # For items up to 1 KB in size,
+    # one WCU can perform one standard write request per second
+
+    mean_item_size = _mean_item_size_bytes(desires, extra_model_arguments)
+
+    rounded_wcus_per_item = math.ceil(max(1.0, mean_item_size / 1024))
+
+    transactional_write_percent: float = extra_model_arguments.get(
+        "transactional_write_percent", 0
+    )
+    assert transactional_write_percent >= 0.0
+    baseline_wcus_non_transactional = (
+        desires.query_pattern.estimated_write_per_second.mid
+        * (1 - transactional_write_percent)
+        * rounded_wcus_per_item
+    )
+    baseline_wcus_transactional = (
+        desires.query_pattern.estimated_write_per_second.mid
+        * transactional_write_percent
+        * 2
+        * rounded_wcus_per_item
+    )
+    # we do not break down reserved vs non-reserved
+    total_baseline_wcus = math.ceil(
+        baseline_wcus_non_transactional + baseline_wcus_transactional
+    )
+
+    # 8760 hours in a year
+    total_baseline_wcus_hours = 8760 * total_baseline_wcus
+    number_of_regions = _get_num_regions(extra_model_arguments)
+    dynamo_service_standard = _get_dynamo_standard(context, extra_model_arguments)
+
+    # for global replication each region will additionally consume rWCUs
+    total_annual_cost_wcu = round(
+        (
+            number_of_regions
```

Comment:
Are rWCUs priced at the same thing as local WCU? Otherwise we want num_regions - 1 right? From https://aws.amazon.com/dynamodb/pricing/provisioned/ I think global writes are more expensive than local writes (0.00065 vs 0.000975 if I'm reading that right)